dex/src/main/scala/com/wavesplatform/dex/Application.scala

```diff
@@ -325,47 +325,69 @@ class Application(settings: MatcherSettings, config: Config)(implicit val actorS
-      p => Option(orderBooks.get()) flatMap (_ get p),
-      orderBookHttpInfo,
-      getActualTickSize = { assetPair =>
-        getAndCacheDecimals(assetPair.amountAsset)
-          .product(getAndCacheDecimals(assetPair.priceAsset))
-          .map { case (amountAssetDecimals, priceAssetDecimals) =>
-            matchingRulesCache.getDenormalizedRuleForNextOrder(assetPair, lastProcessedOffset, amountAssetDecimals, priceAssetDecimals).tickSize
-          }
-      },
-      orderValidation,
-      settings,
-      () => status,
-      orderDb,
-      () => lastProcessedOffset,
-      () => matcherQueue.lastOffset,
-      ExchangeTransactionCreator.getAdditionalFeeForScript(hasMatcherAccountScript),
-      maybeApiKeyHash,
-      rateCache,
-      validatedAllowedOrderVersions = () => {
-        Future
-          .sequence {
-            settings.allowedOrderVersions.map(OrderValidator.checkOrderVersion(_, wavesBlockchainAsyncClient.isFeatureActivated).value)
-          }
-          .map(_.collect { case Right(version) => version })
-      },
-      () => orderFeeSettingsCache.getSettingsForOffset(lastProcessedOffset + 1),
-      externalClientDirectoryRef,
-      getAndCacheDescription
-    )
+    new PlaceRoute(pairBuilder, addressDirectoryRef, orderValidation, settings, () => status, maybeApiKeyHash)
   }
 
-  private val httpApiRouteV1 = MatcherApiRouteV1(pairBuilder, orderBookHttpInfo, () => status, maybeApiKeyHash)
+  private val cancelRoute = new CancelRoute(pairBuilder, addressDirectoryRef, () => status, orderDb, maybeApiKeyHash, settings)
+  private val ratesRoute = new RatesRoute(pairBuilder, () => status, maybeApiKeyHash, rateCache, externalClientDirectoryRef)
+  private val historyRoute = new HistoryRoute(pairBuilder, addressDirectoryRef, () => status, maybeApiKeyHash, settings)
+  private val balancesRoute = new BalancesRoute(pairBuilder, addressDirectoryRef, () => status, maybeApiKeyHash, settings)
+  private val transactionsRoute = new TransactionsRoute(() => status, orderDb, maybeApiKeyHash)
+
+  private val debugRoute = new DebugRoute(
+    config,
+    orderBookDirectoryActorRef,
+    addressDirectoryRef,
+    wavesBlockchainAsyncClient.status(),
+    () => status,
+    () => lastProcessedOffset,
+    () => matcherQueue.lastOffset,
+    maybeApiKeyHash,
+    settings
+  )
+
+  private val marketsRoute = new MarketRoute(
+    addressDirectoryRef,
+    orderDb,
+    pairBuilder,
+    matcherPublicKey,
+    orderBookDirectoryActorRef,
+    matcherQueue.store,
+    p => Option(orderBooks.get()) flatMap (_ get p),
+    orderBookHttpInfo,
+    getActualTickSize = { assetPair =>
+      getAndCacheDecimals(assetPair.amountAsset)
+        .product(getAndCacheDecimals(assetPair.priceAsset))
+        .map { case (amountAssetDecimals, priceAssetDecimals) =>
+          matchingRulesCache.getDenormalizedRuleForNextOrder(assetPair, lastProcessedOffset, amountAssetDecimals, priceAssetDecimals).tickSize
+        }
+    },
+    settings,
```

Comment:
Let's create `Settings` in `MarketRoute` companion with `responseTimeout` and `orderRestrictions` and create and pass the instance as the first argument.